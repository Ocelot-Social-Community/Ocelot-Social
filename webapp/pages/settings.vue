<template>
  <div>
    <ds-space margin="small">
      <ds-heading tag="h1">{{ $t('settings.name') }}</ds-heading>
    </ds-space>
    <ds-space margin="large" />
    <ds-flex gutter="small">
      <div class="menu-container">
        <ds-menu :routes="routes" :is-exact="() => true" />
      </div>
      <div class="settings-content" id="settings-content">
        <transition name="slide-up" appear>
          <nuxt-child />
        </transition>
      </div>
    </ds-flex>
  </div>
</template>

<script>
export default {
  computed: {
    routes() {
      return [
        {
          name: this.$t('settings.data.name'),
          path: `/settings`,
        },
        {
          name: this.$t('settings.email.name'),
          path: `/settings/my-email-address`,
        },
        {
          name: this.$t('settings.security.name'),
          path: `/settings/security`,
        },
        {
          name: this.$t('settings.privacy.name'),
          path: '/settings/privacy',
        },
        {
          name: this.$t('settings.social-media.name'),
          path: `/settings/my-social-media`,
        },
        {
          name: this.$t('settings.muted-users.name'),
          path: `/settings/muted-users`,
        },
        {
          name: this.$t('settings.blocked-users.name'),
          path: `/settings/blocked-users`,
        },
        {
          name: this.$t('settings.embeds.name'),
          path: `/settings/embeds`,
        },
        {
          name: this.$t('settings.notifications.name'),
          path: '/settings/notifications',
        },
        {
          name: this.$t('settings.download.name'),
          path: `/settings/data-download`,
        },
        {
          name: this.$t('settings.deleteUserAccount.name'),
          path: `/settings/delete-account`,
        },
        // TODO implement
        /* {
            name: this.$t('settings.invites.name'),
            path: `/settings/invites`
            }, */
        // TODO implement
        /* {
            name: this.$t('settings.organizations.name'),
            path: `/settings/my-organizations`
            }, */
        // TODO implement
        /* {
            name: this.$t('settings.languages.name'),
            path: `/settings/languages`
            },
            } */
      ]
    },
  },
}
</script>

<style scoped>
.menu-container {
  width: 100%
}
.settings-content {
  flex: 1;
  padding: 0 24px;
  margin-top: 32px;
}

@media screen and (min-width: 600px) {
  .settings-content {
    margin-top: 0;
  }

  .menu-container {
    width: 200px;
  }
}
</style>
