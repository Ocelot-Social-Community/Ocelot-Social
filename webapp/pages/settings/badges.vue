<template>
  <base-card>
    <h2 class="title">{{ $t('settings.badges.name') }}</h2>
    <ds-space margin-bottom="big">
      <div class="presenterContainer">
        <badges
          :badges="[currentUser.badgeVerification, ...selectedBadges]"
          :selection-mode="true"
          :scale="2"
          @badge-selected="handleBadgeSlotSelection"
          ref="badgesComponent"
        />
      </div>

      <div v-if="selectedBadgeIndex !== null && !isEmptySlot" class="badge-actions">
        <base-button @click="removeBadgeFromSlot" class="remove-button">
          {{ $t('settings.badges.remove') }}
        </base-button>
      </div>

      <div v-if="selectedBadgeIndex !== null && isEmptySlot" class="selection-info">
        <badge-selection
          :badges="availableBadges"
          @badge-selected="assignBadgeToSlot"
          ref="badgeSelection"
        />
      </div>
    </ds-space>
  </base-card>
</template>

<script>
import { mapGetters, mapMutations } from 'vuex'
import { setTrophyBadgeSelected } from '~/graphql/User'
import scrollToContent from './scroll-to-content.js'
import Badges from '../../components/Badges.vue'
import BadgeSelection from '../../components/BadgeSelection.vue'

export default {
  components: { BadgeSelection, Badges },
  mixins: [scrollToContent],
  data() {
    return {
      selectedBadgeIndex: null,
    }
  },
  computed: {
    ...mapGetters({
      currentUser: 'auth/user',
    }),
    selectedBadges() {
      return this.currentUser.badgeTrophiesSelected
    },
    availableBadges() {
      return this.currentUser.badgeTrophiesUnused
    },
    isEmptySlot() {
      console.log('selectedBadgeIndex', this.selectedBadgeIndex)
      console.log('selectedBadges', this.selectedBadges)
      return this.selectedBadges[this.selectedBadgeIndex]?.isDefault ?? true
    },
  },
  created() {
    this.userBadges = [...(this.currentUser.badgeTrophiesSelected || [])]
  },
  methods: {
    ...mapMutations({
      setCurrentUser: 'auth/SET_USER',
    }),
    handleBadgeSlotSelection(index) {
      if (index === 0) {
        this.$toast.info(
          this.$t('settings.badges.verification') ||
            'Dies ist deine Verifikations-Badge und kann nicht geÃ¤ndert werden.',
        )
        this.$refs.badgesComponent.resetSelection()
        return
      }

      this.selectedBadgeIndex = index === null ? null : index - 1 // The first badge in badges component is the verification badge
    },
    async setSlot(badge, slot) {
      await this.$apollo.mutate({
        mutation: setTrophyBadgeSelected,
        variables: {
          badgeId: badge?.id ?? null,
          slot,
        },
        update: (_, { data: { setTrophyBadgeSelected } }) => {
          const { badgeTrophiesSelected, badgeTrophiesUnused } = setTrophyBadgeSelected
          this.setCurrentUser({
            ...this.currentUser,
            badgeTrophiesSelected,
            badgeTrophiesUnused,
          })
        },
      })
    },
    async assignBadgeToSlot(badge) {
      if (!badge || this.selectedBadgeIndex === null) {
        return
      }

      await this.setSlot(badge, this.selectedBadgeIndex)

      this.$toast.success(this.$t('settings.badges.success-update'))

      if (this.$refs.badgeSelection && this.$refs.badgeSelection.resetSelection) {
        this.$refs.badgeSelection.resetSelection()
      }
      this.$refs.badgesComponent.resetSelection()
      this.selectedBadgeIndex = null
    },
    async removeBadgeFromSlot() {
      if (this.selectedBadgeIndex === null) return

      await this.setSlot(null, this.selectedBadgeIndex)

      this.$refs.badgesComponent.resetSelection()
      this.selectedBadgeIndex = null
    },
  },
}
</script>

<style lang="scss">
.presenterContainer {
  padding-top: 50px;
  min-height: 250px;
}

.badge-actions {
  margin-top: 20px;
  display: flex;
  justify-content: center;

  .remove-button {
    margin-top: 8px;
  }
}

.selection-info {
  margin-top: 20px;
  padding: 16px;
}
</style>
