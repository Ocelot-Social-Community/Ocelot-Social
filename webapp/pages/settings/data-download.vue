<template>
  <base-card>
    <h2 class="title">{{ $t('settings.download.name') }}</h2>
    <base-button
      @click="onClick(jsonData)"
      icon="download"
      secondary
      filled
      :disabled="loading"
      :loading="loading"
    >
      {{ $t('settings.download.json') }}
    </base-button>
    <ds-space margin="large" />
    <ds-text>{{ $t('settings.download.description') }}</ds-text>
    <ds-space margin="large" />
    <base-card v-for="image in imageList" :key="image.key">
      <a :href="image.url" target="_blank" rel="noopener noreferrer">{{ image.title }}</a>
      <ds-space margin="xxx-small" />
    </base-card>
  </base-card>
</template>

<script>
import { mapGetters } from 'vuex'
import { userDataQuery } from '~/graphql/User'
import BaseButton from '~/components/_new/generic/BaseButton/BaseButton.vue'
import isEmpty from 'lodash/isEmpty'
import scrollToContent from './scroll-to-content.js'

export default {
  mixins: [scrollToContent],
  components: {
    BaseButton,
  },
  data() {
    return {
      userData: {},
      loading: true,
      imageList: [],
    }
  },
  computed: {
    ...mapGetters({
      user: 'auth/user',
    }),
    jsonData() {
      return { data: JSON.stringify(this.userData, null, 2), type: 'json' }
    },
  },
  methods: {
    onClick(method) {
      var fileURL = window.URL.createObjectURL(new Blob([method.data]))
      var fileLink = document.createElement('a')
      fileLink.href = fileURL
      fileLink.setAttribute('download', 'userData.' + method.type)
      document.body.appendChild(fileLink)
      fileLink.click()
    },
  },
  apollo: {
    queryUserData: {
      query() {
        return userDataQuery()
      },
      variables() {
        return { id: this.user.id }
      },
      update({ userData }) {
        this.userData = userData
        this.loading = false
        if (isEmpty(this.userData)) return null
        const userId = this.userData.user.id
        if (isEmpty(userId)) return null
        this.imageList = this.userData.posts
          .filter((post) => post.author.id === userId && post.image)
          .map((post) => {
            const obj = {}
            obj.key = post.id
            obj.url = post.image.url
            obj.title = post.title
            return obj
          })
      },
      fetchPolicy: 'cache-and-network',
    },
  },
}
</script>
