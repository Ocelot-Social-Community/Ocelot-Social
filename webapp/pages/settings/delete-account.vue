<template>
  <delete-data />
</template>

<script>
import DeleteData from '~/components/DeleteData/DeleteData.vue'
import scrollToContent from './scroll-to-content.js'

export default {
  mixins: [scrollToContent],
  components: {
    DeleteData,
  },
}
</script>
