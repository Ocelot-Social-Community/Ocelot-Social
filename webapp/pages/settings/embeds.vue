<template>
  <base-card>
    <h2 class="title">{{ $t('settings.embeds.name') }}</h2>
    <ds-section>
      <ds-text>
        {{ $t('settings.embeds.status.description') }}
        <ds-text bold>
          <template v-if="disabled">
            {{ $t(`settings.embeds.status.disabled.on`) }}
          </template>
          <template v-else>
            {{ $t(`settings.embeds.status.disabled.off`) }}
          </template>
        </ds-text>
        .
      </ds-text>
      <ds-text>
        {{ $t('settings.embeds.status.change.question') }}
      </ds-text>
      <base-button @click="submit" :filled="!disabled" :disabled="!disabled" style="margin-right: 5px;">
        {{ $t('settings.embeds.status.change.deny') }}
      </base-button>
      <base-button @click="submit" :filled="disabled" :disabled="disabled">
        {{ $t('settings.embeds.status.change.allow') }}
      </base-button>

      <p>{{ $t('settings.embeds.info-description') }}</p>
      <ul>
        <li v-for="provider in providers" :key="provider.provider_name">
          {{ provider.provider_name }},
          <small>{{ provider.provider_url }}</small>
        </li>
      </ul>
    </ds-section>
  </base-card>
</template>

<script>
import axios from 'axios'
import { mapGetters, mapMutations } from 'vuex'
import { updateUserMutation } from '~/graphql/User.js'

export default {
  head() {
    return {
      title: this.$t('settings.embeds.name'),
    }
  },
  computed: {
    ...mapGetters({
      currentUser: 'auth/user',
    }),
  },
  data() {
    return {
      disabled: null,
      print: true,
      providers: [],
    }
  },
  mounted() {
    axios.get('/api/providers.json').then((response) => {
      this.providers = response.data
    })
    if (this.currentUser.allowEmbedIframes) this.disabled = this.currentUser.allowEmbedIframes
  },
  methods: {
    ...mapMutations({
      setCurrentUser: 'auth/SET_USER',
    }),
    async submit() {
      try {
        await this.$apollo.mutate({
          mutation: updateUserMutation(),
          variables: {
            id: this.currentUser.id,
            allowEmbedIframes: !this.disabled,
          },
          update: (store, { data: { UpdateUser } }) => {
            const { allowEmbedIframes } = UpdateUser
            this.setCurrentUser({
              ...this.currentUser,
              allowEmbedIframes,
            })
          },
        })
        this.disabled = !this.disabled
      } catch (err) {
        this.$toast.error(err.message)
      }
    },
  },
}
</script>
