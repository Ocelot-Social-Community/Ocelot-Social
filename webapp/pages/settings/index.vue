<template>
  <ds-form class="settings-form" v-model="formData" :schema="formSchema" @submit="submit">
    <template #default="{ errors }">
      <base-card>
        <h2 class="title">{{ $t('settings.data.name') }}</h2>
        <ds-input
          id="name"
          model="name"
          icon="user"
          :label="
            $env.ASK_FOR_REAL_NAME
              ? $t('settings.data.realNamePlease')
              : $t('settings.data.labelName')
          "
          :placeholder="$t('settings.data.namePlaceholder')"
        />
        <ds-input id="slug" model="slug" icon="at" :label="$t('settings.data.labelSlug')" />
        <location-select
          class="location-selet"
          v-model="formData.locationName"
          :canBeCleared="!$env.REQUIRE_LOCATION"
        />
        <!-- eslint-enable vue/use-v-on-exact -->
        <ds-input
          id="about"
          model="about"
          type="textarea"
          rows="3"
          :label="$t('settings.data.labelBio')"
          :placeholder="$t('settings.data.labelBio')"
        />
        <base-button icon="check" :disabled="errors" type="submit" :loading="loadingData" filled>
          {{ $t('actions.save') }}
        </base-button>
      </base-card>
    </template>
  </ds-form>
</template>

<script>
import { mapGetters, mapMutations } from 'vuex'
import UniqueSlugForm from '~/components/utils/UniqueSlugForm'
import LocationSelect from '~/components/Select/LocationSelect'
import { updateUserMutation } from '~/graphql/User'
import scrollToContent from './scroll-to-content.js'

export default {
  mixins: [scrollToContent],
  name: 'NewsFeed',
  components: {
    LocationSelect,
  },
  data() {
    return {
      cities: [],
      loadingData: false,
      loadingGeo: false,
      formData: {
        name: '',
        slug: '',
        about: '',
        locationName: '',
      },
    }
  },
  mounted() {
    this.formData.name = this.currentUser.name
    this.formData.slug = this.currentUser.slug
    this.formData.about = this.currentUser.about
    this.formData.locationName = this.currentUser.locationName || ''
  },
  computed: {
    ...mapGetters({
      currentUser: 'auth/user',
    }),
    formSchema() {
      const uniqueSlugForm = UniqueSlugForm({
        apollo: this.$apollo,
        currentUser: this.currentUser,
        translate: this.$t,
      })
      return {
        locationName: { required: this.$env.REQUIRE_LOCATION },
        name: { required: true, min: 3 },
        ...uniqueSlugForm.formSchema,
      }
    },
  },
  methods: {
    ...mapMutations({
      setCurrentUser: 'auth/SET_USER',
    }),
    async submit() {
      this.loadingData = true
      const { name, slug, about } = this.formData
      let { locationName } = this.formData || this.currentUser
      locationName = locationName && (locationName.label || locationName)
      try {
        await this.$apollo.mutate({
          mutation: updateUserMutation(),
          variables: {
            id: this.currentUser.id,
            name,
            slug,
            locationName,
            about,
          },
          update: (store, { data: { UpdateUser } }) => {
            this.setCurrentUser({
              ...this.currentUser,
              ...UpdateUser,
            })
          },
        })
        this.$toast.success(this.$t('settings.data.success'))
      } catch (err) {
        this.$toast.error(err.message)
      } finally {
        this.loadingData = false
      }
    },
  },
}
</script>

<style lang="scss">
.location-hint {
  margin-top: -$space-x-small - $space-xxx-small - $space-xxx-small;
}

.location-selet {
  margin-bottom: $space-small;
}
</style>
