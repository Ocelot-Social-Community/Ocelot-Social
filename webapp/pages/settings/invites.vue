<template>
  <base-card>
    <h2 class="title">{{ $t('settings.invites.name') }}</h2>
    <empty icon="tasks" message="Coming Soonâ€¦" />
  </base-card>
</template>

<script>
import Empty from '~/components/Empty/Empty'

export default {
  components: {
    Empty,
  },
}
</script>
