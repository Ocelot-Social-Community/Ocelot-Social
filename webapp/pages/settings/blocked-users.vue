<template>
  <div>
    <ds-space>
      <base-card>
        <h2 class="title">{{ $t('settings.blocked-users.name') }}</h2>
        <ds-text>
          {{ $t('settings.blocked-users.explanation.intro') }}
        </ds-text>
        <ds-list>
          <ds-list-item>
            {{ $t('settings.blocked-users.explanation.your-perspective') }}
          </ds-list-item>
          <ds-list-item>
            {{ $t('settings.blocked-users.explanation.their-perspective') }}
          </ds-list-item>
          <ds-list-item>
            {{ $t('settings.blocked-users.explanation.notifications') }}
          </ds-list-item>
        </ds-list>
      </base-card>
    </ds-space>
    <base-card v-if="blockedUsers && blockedUsers.length">
      <ds-table :data="blockedUsers" :fields="fields" condensed>
        <template #avatar="scope">
          <nuxt-link
            :to="{
              name: 'profile-id-slug',
              params: { id: scope.row.id, slug: scope.row.slug },
            }"
          >
            <profile-avatar :profile="scope.row" size="small" />
          </nuxt-link>
        </template>
        <template #name="scope">
          <nuxt-link
            :to="{
              name: 'profile-id-slug',
              params: { id: scope.row.id, slug: scope.row.slug },
            }"
          >
            <b>{{ scope.row.name | truncate(20) }}</b>
          </nuxt-link>
        </template>
        <template #slug="scope">
          <nuxt-link
            :to="{
              name: 'profile-id-slug',
              params: { id: scope.row.id, slug: scope.row.slug },
            }"
          >
            <b>{{ scope.row.slug | truncate(20) }}</b>
          </nuxt-link>
        </template>

        <template #unblockUser="scope">
          <base-button circle size="small" @click="unblockUser(scope)" icon="user-plus" />
        </template>
      </ds-table>
    </base-card>
    <base-card v-else>
      <ds-space>
        <ds-placeholder>
          {{ $t('settings.blocked-users.empty') }}
        </ds-placeholder>
      </ds-space>
      <ds-space>
        <ds-text align="center">
          {{ $t('settings.blocked-users.how-to') }}
        </ds-text>
      </ds-space>
    </base-card>
  </div>
</template>

<script>
import { blockedUsers, unblockUser } from '~/graphql/settings/BlockedUsers'
import ProfileAvatar from '~/components/_new/generic/ProfileAvatar/ProfileAvatar'
import scrollToContent from './scroll-to-content.js'

export default {
  mixins: [scrollToContent],
  components: {
    ProfileAvatar,
  },
  data() {
    return {
      blockedUsers: [],
    }
  },
  computed: {
    fields() {
      return {
        avatar: '',
        name: this.$t('settings.blocked-users.columns.name'),
        slug: this.$t('settings.blocked-users.columns.slug'),
        unblockUser: this.$t('settings.blocked-users.columns.unblock'),
      }
    },
  },
  apollo: {
    blockedUsers: { query: blockedUsers, fetchPolicy: 'cache-and-network' },
  },
  methods: {
    async unblockUser(user) {
      await this.$apollo.mutate({
        mutation: unblockUser(),
        variables: { id: user.row.id },
      })
      this.$apollo.queries.blockedUsers.refetch()
      const { name } = user.row
      this.$toast.success(this.$t('settings.blocked-users.unblocked', { name }))
    },
  },
}
</script>

<style lang="scss">
.ds-table-col {
  vertical-align: middle;
}
</style>
