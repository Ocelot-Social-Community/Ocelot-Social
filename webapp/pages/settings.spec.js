import { mount } from '@vue/test-utils'
import settings from './settings.vue'

const localVue = global.localVue

const stubs = {
  'nuxt-child': true,
}

describe('settings.vue', () => {
  let wrapper
  let mocks

  beforeEach(() => {
    mocks = {
      $t: jest.fn(),
    }
  })

  describe('mount', () => {
    const Wrapper = () => {
      return mount(settings, {
        mocks,
        localVue,
        stubs,
      })
    }

    beforeEach(() => {
      wrapper = Wrapper()
    })

    it('renders', () => {
      expect(wrapper.element.tagName).toBe('DIV')
    })
  })
})
