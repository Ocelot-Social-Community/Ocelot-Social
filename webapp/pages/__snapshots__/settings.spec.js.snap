// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings.vue given badges are disabled renders 1`] = `
<div>
  <div>
    <div
      class="ds-space"
      style="margin-top: 16px; margin-bottom: 16px;"
    >
      <h1
        class="ds-heading ds-heading-h1"
      >
        
      </h1>
    </div>
     
    <div
      class="ds-space"
      style="margin-top: 32px; margin-bottom: 32px;"
    />
     
    <div
      class="ds-flex"
      style="margin-left: -8px; margin-right: -8px;"
    >
      <div
        class="menu-container"
      >
        <nav
          class="ds-menu"
        >
          <ul
            class="ds-menu-list"
          >
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/my-email-address"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/security"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/privacy"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/my-social-media"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/muted-users"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/blocked-users"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/embeds"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/notifications"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/data-download"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/delete-account"
              >
                
      
    
              </a>
              <!---->
            </li>
          </ul>
        </nav>
      </div>
       
      <div
        class="settings-content"
        id="settings-content"
      >
        <transition-stub
          appear="true"
          name="slide-up"
        >
          <nuxt-child-stub />
        </transition-stub>
      </div>
    </div>
  </div>
</div>
`;

exports[`settings.vue given badges are enabled renders 1`] = `
<div>
  <div>
    <div
      class="ds-space"
      style="margin-top: 16px; margin-bottom: 16px;"
    >
      <h1
        class="ds-heading ds-heading-h1"
      >
        
      </h1>
    </div>
     
    <div
      class="ds-space"
      style="margin-top: 32px; margin-bottom: 32px;"
    />
     
    <div
      class="ds-flex"
      style="margin-left: -8px; margin-right: -8px;"
    >
      <div
        class="menu-container"
      >
        <nav
          class="ds-menu"
        >
          <ul
            class="ds-menu-list"
          >
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/my-email-address"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/badges"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/security"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/privacy"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/my-social-media"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/muted-users"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/blocked-users"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/embeds"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/notifications"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/data-download"
              >
                
      
    
              </a>
              <!---->
            </li>
            <li
              class="ds-menu-item ds-menu-item-level-0"
            >
              <a
                class="ds-menu-item-link"
                exact="true"
                href="/settings/delete-account"
              >
                
      
    
              </a>
              <!---->
            </li>
          </ul>
        </nav>
      </div>
       
      <div
        class="settings-content"
        id="settings-content"
      >
        <transition-stub
          appear="true"
          name="slide-up"
        >
          <nuxt-child-stub />
        </transition-stub>
      </div>
    </div>
  </div>
</div>
`;
